woocommerce/woocommerce
# Try: Product forms as custom post types

### Changes proposed in this Pull Request:

This PR attempts to separate the product forms into their own custom post type.  The main advantages of this are:

* Separation of concerns for product forms and product templates, which are often conflated
* Allows editing forms by developers or merchants for a more customized experience
* Allow usage of GB APIs.  This allows us to use the block editor for editing product forms, retrieval of product forms via the posts endpoint, block and hook processing of forms automatically on all forms

**Product editing**
<img width="993" alt="Screenshot 2024-03-26 at 9 33 10 PM" src="https://github.com/woocommerce/woocommerce/assets/10561050/c03af73e-ae1b-4666-8197-0483250b22d8">
**Product form editing**
<img width="1140" alt="Screenshot 2024-03-26 at 9 31 12 PM" src="https://github.com/woocommerce/woocommerce/assets/10561050/e1a9222a-5d41-43db-802e-577e82afe343">




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Navigate to Product Form -> Add new
2. Create a title for your form and add some fields (only checkbox, regular price, name, and section have been updated to work in this PR)
3. Save your form and optionally create another
4. Navigate to Products -> Add new
5. See your form has been used to populate the fields in the product editor
6. If you created multiple forms, switch between forms using the select dropdown
7. Add some information and save your product
8. Note that the editor saves and works for hydrating edited product data as well



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
